client/src/common/components/Activities/components/activity.jsx

```diff
@@ -0,0 +1,110 @@
+import React, { Fragment, PureComponent } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import { Link } from 'react-router-dom';
+
+import { dateFromString } from 'common/utils/helpers';
+import UserIconPlaceholder from 'assets/images/user.svg';
+import { UserIcon } from 'assets/images/icons';
+
+class Activity extends PureComponent {
+  state = {
+    isAvatarError: false
+  };
+
+  handleAvatarError = () => this.setState({ isAvatarError: true });
+
+  renderCohortInfo = () => {
+    const {
+      activity: { cohort }
+    } = this.props;
+
+    if (cohort) {
+      const { cohortId, cohortName } = cohort;
+      return (
+        <Fragment>
+          {' in '}
+          <Link className="activity__link" to={`/cohort/${cohortId}`}>
+            {cohortName}
+          </Link>
+          {' cohort'}
+        </Fragment>
+      );
+    }
+  };
+
+  renderListInfo = () => {
+    const {
+      activity: { list }
+    } = this.props;
+
+    if (list) {
+      const { listId, listName } = list;
+      return (
+        <Fragment>
+          <Link className="activity__link" to={`/sack/${listId}`}>
+            {listName}
+          </Link>
+        </Fragment>
+      );
+    }
+  };
+
+  render() {
+    const {
+      activity: {
+        actor: { actorAvatarUrl, actorName },
+        createdAt,
+        item: { itemName }
+      }
+    } = this.props;
+    const { isAvatarError } = this.state;
+    const date = dateFromString(createdAt);
+
+    return (
+      <div className="activity">
+        <div
```

Comment:
could this div be extracted to separate `Avatar` component with all error handling functionality?